### PR TITLE
Add fixtures data mocks to acceptance tests

### PR DESCRIPTION
**Pull request by andrew-gardener**
Test data is now initialized based on the "I'm an x" steps which moved into separate step files by user type so that it is easier to keep track of. These steps will load a fixture file which will create the necessary test data using factories.

http_backend_mocks has been refactored putting all the api request mocks into one module. This module contains a storage variable with some default data that will be extended by the fixture data when initialized.

related to #261

### CODE REVIEW

**Comment by andrew-gardener** [comment]
I need to fix the object.assign issue (doesn't happen locally)

File: acj/static/test/config/protractor_cucumber.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 exports.config = {
     //seleniumAddress: 'http://localhost:4444/wd/hub',
-	seleniumServerJar: '../../../../node_modules/protractor/selenium/selenium-server-standalone-2.47.1.jar',
+	//seleniumServerJar: '../../../../node_modules/protractor/selenium/selenium-server-standalone-2.47.1.jar',
```
Comment (xcompass) [inline acj/static/test/config/protractor_cucumber.js]: You don't need to specify selenium jar in new version?

Comment (andrew-gardener) [inline acj/static/test/config/protractor_cucumber.js]: The jar file no longer exists in the location referenced in the newer version. Protractor seems to auto load it itself.

**Comment by xcompass** [comment]
👍

**Pull request merged**